syntax = "proto3";
package task_cascadence.ume.protos;

import "google/protobuf/timestamp.proto";

message TaskSpec {
  string id = 1;
  string name = 2;
  string description = 3;
  string user_hash = 4;
  string user_id = 5;
  optional string group_id = 6;
}

message TaskRun {
  TaskSpec spec = 1;
  string run_id = 2;
  string status = 3;
  google.protobuf.Timestamp started_at = 4;
  google.protobuf.Timestamp finished_at = 5;
  string user_hash = 6;
  string user_id = 7;
  optional string group_id = 8;
}

message TaskPointer {
  string run_id = 1;
  string user_hash = 2;
  string user_id = 3;
  optional string group_id = 4;
}

message PointerUpdate {
  string task_name = 1;
  string run_id = 2;
  string user_hash = 3;
  string user_id = 4;
  optional string group_id = 5;
}

message TaskNote {
  string task_name = 1;
  string run_id = 2;
  string note = 3;
  string user_hash = 4;
  string user_id = 5;
  optional string group_id = 6;
}

message IdeaSeed {
  string text = 1;
  string user_hash = 2;
  string user_id = 3;
  optional string group_id = 4;
}

message StageUpdate {
  string task_name = 1;
  string stage = 2;
  string user_hash = 3;
  string user_id = 4;
  optional string group_id = 5;
}
